odoh_server.go

```diff
@@ -72,28 +77,30 @@ type odohServer struct {
 	DOHURI    string
 }
 
-func (s odohServer) queryHandler(w http.ResponseWriter, r *http.Request) {
+func (s odohServer) proxyHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Now that these are separate endpoints, this should just forward directly to the internal proxy handler:

```go
func (s odohServer) proxyHandler(w http.ResponseWriter, r *http.Request) {
   s.proxy.proxyQueryHandler(w, r)
}
```

That function checks for target name and path internally.

Reply:
I decided to remove the `proxyHandler` and `queryHandler` and directly call the internal functions.

Reply:
👍 Yep, that's better!